astropy/coordinates/sky_coordinate.py

```diff
@@ -624,6 +624,10 @@ def transform_to(self, frame, merge_attributes=True):
                      set(frame_kwargs.keys())):
             frame_kwargs.pop(attr)
 
+        # Ensure transformed coordinates have representation and differential types
+        # as set by the destination frame
+        new_coord.set_representation_cls(frame.default_representation, frame.default_differential)
+
```

Comment:
I'm a bit confused here.

- Setting the new coordinates `representation_type` and `differential_type` to the **default of the new frame** makes some tests fail.
- Setting the new coordinates `representation_type` and `differential_type` to the **values of the original frame** makes some other tests fail.

Therefore, I understand that the logic is a bit more complex for `SkyCoord.transform_to` and that the magic happens at the "attributes merging" stage. Am I right?